### PR TITLE
Bunch of new extractors, other stuff I pr'd on ytdl

### PR DESCRIPTION
**Pull request by mkg20001**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Just found this fork after seeing ytdl being a little bit inactive, decided to drop everything I pr'd there in here

If that is a problem I can also make seperate PRs, but would prefer not to

### CODE REVIEW

**Comment by pukkandan** [comment]
Seperate PRs would have been ideal, but I can work with this.

Could you link the equivalent ytdl PRs for easy reference?

**Comment by mkg20001** [comment]
- https://github.com/ytdl-org/youtube-dl/pull/29621 vivo
- https://github.com/ytdl-org/youtube-dl/pull/29618 vidlox
- https://github.com/ytdl-org/youtube-dl/pull/29616 voe

The others haven't been PRd to youtube-dl yet

**Comment by pukkandan** [comment]
I want through voe, vidlox and streamtape. But all of these seem to be websites primarily used for piracy. Even the test links you added to the code are copyright infringing. Such sites will not be added to yt-dlp

https://github.com/ytdl-org/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free

File: yt_dlp/extractor/shared.py [hunk 1]
```diff
@@ -116,7 +116,7 @@ def _extract_title(self, webpage):
 
     def _extract_filesize(self, webpage):
         return parse_filesize(self._search_regex(
-            r'data-type=["\']video["\'][^>]*>Watch.*?<strong>\s*\((.+?)\)',
+            r'<strong>\((\d+\.\d+ [KGM]B)\)</strong>',
```
Comment (pukkandan) [inline yt_dlp/extractor/shared.py:119]: Add a working test case

**Comment by CHJ85** [comment]
These sites are primarily used for piracy. Not that adding them to yt-dlp infringes on any copyright so long as your examples aren't copyright protected. But I understand why @pukkandan wants to be on the safe side.
What I suggest you do @mkg20001 is convert these extractors into plugins and create your own repo called yt-dlp-plugins or something.

**Comment by yinshiyi** [comment]
Should this PR be closed?

**Comment by pelim** [comment]
I created as plugin for voe.sx download https://github.com/pelim/yt-dlp-plugin-voe

**Pull request closed without merging**